### PR TITLE
[#539] Adding commit message checks to ci workflow

### LINKED ISSUE #539: Adding commit message rules to ci
**Is your feature request related to a problem? Please describe.**
This makes sure each commite message should begin with [#issue] or NO-JIRA

**Describe the solution you'd like**

**Describe alternatives you've considered**
Optional. A clear and concise description of any alternative solutions or features you've considered.

**Additional context**
Optional. Add any other context or screenshots about the feature request here.

### PR DESCRIPTION
**Pull request by gaohoward**

### CODE REVIEW

**Comment by gaohoward** [comment]
@brusdev It's ready for review. :)

**Comment by gtully** [comment]
wondering about the NO-JIRA bit. maybe on a PR, that is not an option. Just push the commit with a good message and bypass the PR process if it is the addition of a test or a fix to a spelling mistake in the doc.
otherwise have an issue# and raise a PR. 
if we need no-jira, maybe call it NO-ISSUE

**Comment by gaohoward** [comment]
> wondering about the NO-JIRA bit. maybe on a PR, that is not an option. Just push the commit with a good message and bypass the PR process if it is the addition of a test or a fix to a spelling mistake in the doc. otherwise have an issue# and raise a PR. if we need no-jira, maybe call it NO-ISSUE

Yeah [NO-JIRA] (or more relevant [NO-ISSUE]) is kinda weak in a PR. Direct pushing to main bypassing the check should happen in rare cases. Most changes should go through a PR process in which case forcing each PR with a issue number seems too strict.

File: .github/workflows/merge-rules.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          nlines=-1
+          for (( N=1; N<=${{github.event.pull_request.commits}}; N++ ))
+          do
+            prev_lines=$(($nlines+1))
+            nlines=0
+            while IFS= read -r line
+            do
+              nlines=$((nlines+1))
+              if (( $nlines == $prev_lines+1 )); then
+                if ! [[ "$line" =~ ^\[#([0-9])+\]\ .+$|^(\[NO-ISSUE\]\ ).+$ ]]; then
```
Comment (brusdev) [inline .github/workflows/merge-rules.yml:30]: Are the round brackets required?
```suggestion
                if ! [[ "$line" =~ ^\[#([0-9])+\]\ .+$|^\[NO-ISSUE\]\ .+$ ]]; then
```

Comment (gaohoward) [reply .github/workflows/merge-rules.yml:30]: Indeed they are not necessary. good catch!

**Comment by gaohoward** [comment]
> @gaohoward nice work!!! I think your PR is ready to be merged, could you squash your commits?

Thanks! will do.

**Review by brusdev — APPROVED** [review]

**Pull request merged**